### PR TITLE
refactor(core)!: use case insensitive search for emails

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

Adopt our [new strategy](https://www.notion.so/silverhand/Email-Address-Resolution-99ee44b471c2409884e4ea4d86815b19): store email in case-sensitive, search in case-insensitive

- use `lower()` to check email existency
- enable admin user API to create user with email, and make username optional
- add integration tests of email and password sign-in

## Testing

- [x] sign in demo app with random uppercase or lowercase in the address local part
- [x] integration test

### CODE REVIEW

File: packages/core/src/routes/admin-user.test.ts [hunk 1]
```diff
@@ -113,24 +117,8 @@ describe('adminUserRoutes', () => {
...
-    );
-    await expect(userRequest.post('/users').send({ username, name })).resolves.toHaveProperty(
-      'status',
-      400
-    );
-    await expect(userRequest.post('/users').send({ password, name })).resolves.toHaveProperty(
-      'status',
-      400
-    );
-
```
Comment (darcyYe) [inline packages/core/src/routes/admin-user.test.ts:133]: we may add test cases that ensure we can not call this API without giving `primaryEmail` and `username` at the same time.

**Review by darcyYe — APPROVED** [review]
Overall LGTM

File: packages/core/src/routes/admin-user.test.ts (hunk 1 above)
Comment (gao-sun) [reply packages/core/src/routes/admin-user.test.ts:133]: i think no need for this since it's mgmt API

Comment (gao-sun) [reply packages/core/src/routes/admin-user.test.ts:133]: discussed offline

**Pull request merged**